modules/internal/src/main/scala/server/monixCalls.scala

```diff
@@ -34,41 +34,54 @@ object monixCalls {
 
   import freestyle.rpc.internal.converters._
 
-  def unaryMethod[F[_]: Effect, Req, Res](f: Req => F[Res]): UnaryMethod[Req, Res] =
+  def unaryMethod[F[_]: Effect, Req, Res](
+      f: Req => F[Res],
+      maybeCompression: Option[String]): UnaryMethod[Req, Res] =
     new UnaryMethod[Req, Res] {
-      override def invoke(request: Req, responseObserver: StreamObserver[Res]): Unit =
+      override def invoke(request: Req, responseObserver: StreamObserver[Res]): Unit = {
+        addCompression(responseObserver, maybeCompression)
         Effect[F]
           .runAsync(f(request))(either => IO(completeObserver(responseObserver)(either)))
           .unsafeRunAsync(_ => ())
+      }
     }
 
-  def clientStreamingMethod[F[_]: Effect, Req, Res](f: Observable[Req] => F[Res])(
-      implicit S: Scheduler): ClientStreamingMethod[Req, Res] =
+  def clientStreamingMethod[F[_]: Effect, Req, Res](
+      f: Observable[Req] => F[Res],
+      maybeCompression: Option[String])(implicit S: Scheduler): ClientStreamingMethod[Req, Res] =
     new ClientStreamingMethod[Req, Res] {
 
-      override def invoke(responseObserver: StreamObserver[Res]): StreamObserver[Req] =
+      override def invoke(responseObserver: StreamObserver[Res]): StreamObserver[Req] = {
+        addCompression(responseObserver, maybeCompression)
```

Comment:
I'm not very cool with this approach, I was trying to change the `addConmpression` method result to `F[Unit]` but I'm not sure how to add an effect here, in the `transformStreamObserver` that needs to be applied before anything else. I think I'll give it a second try.